### PR TITLE
Docs Feedback: Fix React Native migration tests

### PR DESCRIPTION
**Pull request by krollins-mdb**
## Pull Request Info

Fixed tests that weren't properly set up. Generated new samples.

### Staged Changes

- [Change an Object Model](https://docs-mongodbcom-staging.corp.mongodb.com/realm/docsworker-xlarge/feedback-fix-rn-migration-tests/sdk/react-native/model-data/change-an-object-model/)

### Review Guidelines

[REVIEWING.md](https://github.com/mongodb/docs-realm/blob/master/REVIEWING.md)

### CODE REVIEW

File: examples/react-native/__tests__/js/realm-database/change-an-object-model-test.jsx [hunk 1]
```diff
@@ -14,32 +14,34 @@ import {render} from '@testing-library/react-native';
 //       "schemaVersion: 5": "schemaVersion: 2",
 //       "oldRealm.schemaVersion < 4": "oldRealm.schemaVersion < 2",
 //       "oldRealm.schemaVersion < 5": "oldRealm.schemaVersion < 2",
-//       "RealmProvider, useRealm": "RealmProvider"
+//        "RealmProvider, useRealm": "RealmProvider"
```
Comment (cbush) [inline examples/react-native/__tests__/js/realm-database/change-an-object-model-test.jsx:17]: Why this change?

File: examples/react-native/__tests__/ts/realm-database/change-an-object-model-test.tsx [hunk 2]
```diff
@@ -216,27 +260,35 @@ describe('Change an Object Model Tests', () => {
...
         </View>
       );
     };
 
     render(<App />);
+
+    // Wait for `RestOfApp` to render and change higher order values.
+    await waitFor(() => {
+      // This test assumes only one object model exists.
+      expect(higherOrderSchema).not.toHaveProperty('properties.lastName');
```
Comment (cbush) [inline examples/react-native/__tests__/ts/realm-database/change-an-object-model-test.tsx:278]: Why not put these in the render function (Bluehawk removed)? Then you could declare higherOrderSchema, etc. at the time of initialization (generally a best practice)

**Review by cbush — APPROVED** [review]
Thanks!

File: examples/react-native/__tests__/js/realm-database/change-an-object-model-test.jsx (hunk 1 above)
Comment (krollins-mdb) [reply examples/react-native/__tests__/js/realm-database/change-an-object-model-test.jsx:17]: Uh, totally accidental. I'll revert that space.

File: examples/react-native/__tests__/ts/realm-database/change-an-object-model-test.tsx (hunk 2 above)
Comment (krollins-mdb) [reply examples/react-native/__tests__/ts/realm-database/change-an-object-model-test.tsx:278]: This is the pattern Ben and I settled on after running into lots of async issues. Can you point me to resources for writing better React Native tests? I've read a few, but most of them don't quite cover what we're doing.

Probably worth re-evaluating the testing philosophy when creating the new test suite for Realm.js v12. For now, I'll keep this, as its the pattern the other React Native tests follow.
